contracts/test/helpers.js

```diff
@@ -151,6 +152,29 @@ chai.Assertion.addMethod("emittedEvent", async function (eventName, args) {
   }
 });
 
+chai.Assertion.addMethod(
+  "revertedWithCustomError",
+  async function (errorSignature) {
+    let txSucceeded = false;
+    try {
+      await this._obj;
+      txSucceeded = true;
+    } catch (e) {
+      const errorHash = keccak256(toUtf8Bytes(errorSignature)).substr(0, 10);
+      chai
+        .expect(e.message)
+        .to.contain(
+          errorHash,
+          `Expected error message with signature ${errorSignature} but another was thrown.`
+        );
+    }
+
+    if (txSucceeded) {
+      chai.expect.fail(`Expected ${errorSignature} error but none was thrown`);
+    }
+  }
+);
+
```

Comment:
🔥🔥🔥
It is super useful and amazing to have it now for testing! 
I remember talking about this recently and that we need to upgrade to ether v6 to have it natively on chai. But you built it, that's amazing !!

Reply:
Yeah I've built sort of a lite version of it :) With new chai we would probably gain support to also test the values of the error parameters. We could decode that out of the error message, but might be out of scope for this PR